### DISCUSSION TITLE
Dropzone: Get both rejected and accepted files on the same callback function

### DISCUSSION THREAD

**Discussion by nicolassanmar** [mantinedev/mantine#2988] [Q&A]
When using `<Dropzone multiple />` I'm trying to show the users the list of the files that got accepted, and the ones that we rejected, because of MIME type, size, etc.
We have the `onDrop` callback function, that gets only the accepted files. We also have the `onReject` callback that fires when files are rejected. 
The issue of using those separately, is that to display the correct information about the last items dropped, might require some complex state management. (If we update the rejected files, and then the user uploads more files, this could lead to inconsistent states)

This is why I'm asking if there is some way of getting both accepted and rejected files on the same callback. That would make the implementation much easier and cleaner

**Comment by rtivital** [comment]
I do not see any issue here, you can store accepted and rejected files in two separate states and then combine them:
```ts
const [accepted] = useState([]);
const [rejected] = useState([]);

const allFiles = [...accepted, ...rejected]
```

**Reply by nicolassanmar** [reply]
But what happens if the user first uploads only invalid files, populating `[rejected]` and then they upload only valid files, populating `[accepted]`?
In each case, only one of `onDrop` and `onReject` methods are fired, making `allFiles` not reflect the last user interaction.

And if we reset the state of the `rejected` files on the `onDrop` function, we may get a race condition on what executes first, `onDrop` or `onReject`
